projects/maximum-subarray/index.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Maximum Subarray
+layout: default
+date: 2020-10-06
+last-modified: 2020-10-12
+featured-image:
+tags: [maximum-subarray]
+authors:
+---
+
+Given an array, the maximum subarray is the subarray which gives the maximum sum possible in the array.
+Kadane's algorithm is a popular algorithm used to find out this maximum sum value.
+
+Note: A subarray is any continuous portion of an array.
+
+For example:
+
+Given array: `[-1, -2, 1, 2, 3, 4, 5, 23]`
+
+Max sum value of a subarray: `38`
+
+Explanation: The subarray `[1, 2, 3, 4, 5, 23]` gives the max sum value. `(1 + 2 + 3 + 4 + 5 + 23 = 38)`
+
+## Requirements
+
+You must write an executable program that accepts an `integer array` on `standard input` via the command line, and outputs the maximum subarray sum value to `standard output`.
+
+Note that the Kadane's algorithm assumes that there is atleast 1 negative integer in the array.
+If there is no negative integer, then the max subarray sum value is the sum of all the elements of the array.
+
+Also note that if the input array is empty, then the output is "The array is empty".
+
+## Testing
+
+Some tests for your program are:
+
+| Description                                                  | Input                     | Output               |
+| :----------------------------------------------------------- | :------------------------ | :------------------- |
+| No input                                                     | []                        | "The array is empty" |
```

Comment:
This table is excellent! Only request would be to specify the error message similar to sorting: https://sample-programs.therenegadecoder.com/projects/bubble-sort/

```suggestion
| No input                                                     | []                        | Usage: please provide a list of at least two integers to sort in the format “1, 2, 3, 4, 5”|
```

Reply:
There should probably be one more error case as well. Also, we should probably use strings instead of lists as input. Otherwise, nice work!

Reply:
I changed the error message and configured it to use strings instead of list. 
What's the one more error case you want me to add @jrg94 ?

Reply:
No input & empty input. They're similar but one is just an empty string:

`maximum-subarray.lang ""`

Reply:
@jrg94 Added the test for empty input as well in the table. Accordingly, added check for it in the python script as well:
https://github.com/TheRenegadeCoder/sample-programs/pull/2056